### PR TITLE
Add File Handling article

### PR DESCRIPTION
**Pull request by tomayac**
Fixes https://github.com/GoogleChrome/web.dev/issues/3954.

Changes proposed in this pull request:

- Add the article.
- This is still pending the OT info.
- [Google doc](https://docs.google.com/document/d/186TZWvonEqxITCp2D7EmUGopzuSxrjNeBGeV_Mvbakc/edit) for reviewing (for @jpmedley).

### CODE REVIEW

File: src/site/content/en/blog/file-handling/index.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+### DevTools support
+
+There is no DevTools support at the time of this writing, but I have filed a
+[feature request](https://bugs.chromium.org/p/chromium/issues/detail?id=1130552) for support to be
+added.
+
+## Demo
+
+I have added File Handling support to [Excalidraw][demo], a cartoon-style drawing app. When you
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:170]:
```suggestion
I have added file handling support to [Excalidraw][demo], a cartoon-style drawing app. When you
```

File: src/site/content/en/blog/file-handling/index.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+create a file with it and store it somewhere on your file system, you can now open the file via a
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:171]:
```suggestion
create a file with it and store it somewhere on your file system, you can open the file via a
```

File: src/site/content/en/blog/file-handling/index.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+double click, or a right click and then selecting "Excalidraw" in the context menu. You can check
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:172]:
```suggestion
double click, or a right click and then select "Excalidraw" in the context menu. You can check
```

File: src/site/content/en/blog/file-handling/index.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+## Security and permissions
+
+The Chrome team has designed and implemented the File Handling API using the core principles defined
+in [Controlling Access to Powerful Web Platform Features][powerful-apis], including user control,
+transparency, and ergonomics.
+
+### File-related challenges
+
+There is a large category of attack vectors that are opened up by allowing websites access to files.
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:201]:
```suggestion
There is a large category of attack vectors that are opened by allowing websites access to files.
```

File: src/site/content/en/blog/file-handling/index.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+There is still a risk that users may unintentionally grant a web application access to a file by
+opening it. However, it is generally understood that opening a file allows the application it is
+opened with to read and/or manipulate that file. Therefore, a user's explicit choice to open a file
+in an installed application, such as via an "Open with…" context menu, can be read as a sufficient
+signal of trust in the application.
+
+### Default handler challenges
+
+The exception to this is where there are no existing applications installed on the host operating
+system capable of handling a given file type. In this case, some host operating systems may
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:219]:
```suggestion
The exception to this is when there are no applications on the host system for a given file
type. In this case, some host operating systems may
```

File: src/site/content/en/blog/file-handling/index.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+automatically promote the newly registered handler to become the default handler for that file type,
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:220]:
```suggestion
automatically promote the newly registered handler to the default handler for that file type,
```

File: src/site/content/en/blog/file-handling/index.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+of that type, it would automatically open in the registered web app. On such host operating systems,
+when the user agent determines that there is no existing default handler for the file type, an
+explicit permission prompt might be necessary to avoid accidentally sending the contents of a file
+to a web application without the user's consent.
+
+### User control
+
+Browsers should not register every site that can handle files as a file handler. Instead,
+registration should be gated behind installation and never happen without explicit user
+confirmation, and especially if a site is to become the default handler.
```
Comment (jpmedley) [inline src/site/content/en/blog/file-handling/index.md:231]:
```suggestion
confirmation, especially if a site is to become the default handler.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/file-handling/index.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+alt: Binders in many colors.
+origin_trial:
+  url:
+---
+
+{% Aside %} The File Handling API is part of the
+[capabilities project](https://web.dev/fugu-status/) and is currently in development. This post will
+be updated as the implementation progresses. {% endAside %}
+
+## What is the File Handling API? {: #what }
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/file-handling/index.md:24]: This comes straight from our [template](https://docs.google.com/document/d/1-yPBCOQcXczf7j9Hn47Ui4Cxqfslkln39rgIWD-AJK0/edit).

File: src/site/content/en/blog/file-handling/index.md [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+step is to let developers declare these very web apps as file handlers for the files their apps can
+create and process. The File Handling API allows you to do exactly this.
+
+### Suggested use cases for the File Handling API {: #use-cases }
+
+Examples of sites that may use this API include:
+
+- Office applications like word editors, spreadsheet apps, and slideshow creators.
+- Graphics editors and drawing tools.
+- Game level design tools.
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/file-handling/index.md:36]: [Video game level editor](https://www.careerexplorer.com/careers/video-game-level-editor/#:~:text=A%20video%20game%20level%20editor%20creates%20the%20interactive%20architecture%20for,and%20variations%20that%20are%20needed.).

File: src/site/content/en/blog/file-handling/index.md [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+
+### Enabling via chrome://flags
+
+To experiment with the File Handling API locally, without an origin trial token, enable the
+`#file-handling-api` flag in `chrome://flags`.
+
+### Enabling support during the origin trial phase
+
+Starting in Chrome&nbsp;⛔️, the File Handling API will be available as an origin trial in Chrome.
+The origin trial is expected to end in Chrome&nbsp;⛔️ (⛔️, 2020).
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/file-handling/index.md:62]: Yepp, versions pending.

File: src/site/content/en/blog/file-handling/index.md [hunk 11]
```diff
@@ -0,0 +1,300 @@
+---
+title: Open files with their proper app
```
Comment (tomayac) [inline src/site/content/en/blog/file-handling/index.md:2]:
```suggestion
title: Give web applications a way to register their ability to handle files
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/file-handling/index.md]: What do you think about the new title?

[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/file-handling/index.md]: It gets the point of view right. I think think you can be pithier: "Let web applications be file handlers".

File: src/site/content/en/blog/file-handling/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/file-handling/index.md]: I'll talk to Pete about it.

File: src/site/content/en/blog/file-handling/index.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/file-handling/index.md]: Oh, a level in a game. Duh. I'm not a gamer.

File: src/site/content/en/blog/file-handling/index.md [hunk 12]
```diff
@@ -0,0 +1,299 @@
+---
+title: Give web applications a way to register their ability to handle files
```
Comment (tomayac) [inline src/site/content/en/blog/file-handling/index.md:2]:
```suggestion
title: Let web applications be file handlers
```

File: src/site/content/en/blog/file-handling/index.md [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+subhead: Register an app as a file handler with the operating system.
+authors:
+  - thomassteiner
+Description: |
+  Register an app as a file handler with the operating system
+  and open files with their proper app.
+date: 2020-09-24
+updated: 2020-09-24
```
Comment (tomayac) [inline src/site/content/en/blog/file-handling/index.md:10]:
```suggestion
date: 2020-09-25
# updated: 2020-09-25
```

File: src/site/content/en/blog/file-handling/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/file-handling/index.md]: This is it :-) Thank you!

**Comment by tomayac** [comment]
@kaycebasques @jpmedley This article can now be merged. We're publishing it as a Dev Trial article for now, since the Origin Trial will take a little longer to be ready. I will of course then update the article once we enter the OT phase.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**